docs/ml/job-tips.asciidoc

```diff
@@ -60,3 +73,21 @@ do not need more than three. If you pick many influencers, the results can be
 overwhelming and there is a small overhead to the analysis.
 
 The job creation wizards in {kib} can suggest which fields to use as influencers.
+
+[[model-memory-limits]]
+==== Model memory limits
+
+For each job, you can optionally specify a `model_memory_limit`, which is the 
+approximate maximum amount of memory resources that are required for analytical 
+processing. Once this limit is approached, data pruning becomes more aggressive. 
+Upon exceeding this limit, new entities are not modeled. For more information 
+about this property, see {ref}/ml-job-resource.html#ml-analysisconfig[Analysis limits].
+
+You can optionally specify the `xpack.ml.max_model_memory_limit` setting, which 
+prevents the creation of jobs with model memory limits greater than this value. 
+For more information, see {ref}/ml-settings.html[Machine learning settings].
+
+If the estimated model memory limit for a job is greater than the model memory 
+limit for the job or the maximum model memory limit for the cluster, the job 
+creation wizards in {kib} generate a warning. 
+//TBD: What is the recommended action?  Increase the limits?
```

Comment:
The default `model_memory_limit` for a new job created in the advanced wizard or directly via the Elasticsearch API is 1GB.  If the ML nodes in the cluster have lots of memory and the only reason for the warning is this default `model_memory_limit`, then the best course of action may simply be to increase the job's `model_memory_limit`.  But before doing this it is worth double-checking that the chosen analysis makes sense.  The default `model_memory_limit` is set relatively low to avoid accidentally creating a job that uses a huge amount of memory.

By default `xpack.ml.max_model_memory_limit` is not set, which means there is no upper bound on the value you can set `model_memory_limit` to for a particular job.  However, if you set `model_memory_limit` too high then it will be impossible to open the resulting job, as ML will not allocate jobs to nodes that have insufficient memory to run them.

If the ML nodes in the cluster do not have sufficient memory to accommodate a job of the estimated size then the only options are:

1. Add bigger ML nodes to the cluster.
2. Accept that the job will hit its memory limit and will not necessarily find all anomalies it would be expected to find.

If the estimated memory requirement is only a little higher than the `model_memory_limit` then the job will probably produce useful results.

In Cloud `xpack.ml.max_model_memory_limit` is set to prevent you creating jobs that could not be allocated to any ML node within the cluster.  If you find that you cannot increase `model_memory_limit` for your ML jobs running in Cloud then the solution is to increase the size of the ML nodes in your cluster.